src/llm/models/report_template.py

```
@@ -129,15 +129,57 @@ def validate_template_type(cls, v):
 
         for provider in v:
-            if provider not in allowed_providers:
-                raise ValueError(f"Unsupported provider: {provider}. Only Gemini is supported in this MVP version.")
+            # 1. Check pattern validity
+            if not pattern.match(provider):
+                # Provide helpful error message based on common issues
+                if provider and provider[0].isupper():
+                    raise ValueError(
+                        f"Invalid provider name '{provider}': Provider names must be lowercase. "
+                        f"Try: '{provider.lower()}'"
+                    )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)